sweep/store.go

```diff
@@ -147,18 +147,16 @@ func migrateTxHashes(tx kvdb.RwTx, txHashesBucket kvdb.RwBucket,
 	return nil
 }
 
-// NotifyPublishTx signals that we are about to publish a tx.
-func (s *sweeperStore) NotifyPublishTx(sweepTx *wire.MsgTx) error {
+// StoreTx stores that we are about to publish a tx.
+func (s *sweeperStore) StoreTx(txid chainhash.Hash) error {
 	return kvdb.Update(s.db, func(tx kvdb.RwTx) error {
 
 		txHashesBucket := tx.ReadWriteBucket(txHashesBucketKey)
 		if txHashesBucket == nil {
 			return errNoTxHashesBucket
 		}
 
-		hash := sweepTx.TxHash()
-
-		return txHashesBucket.Put(hash[:], []byte{})
+		return txHashesBucket.Put(txid[:], []byte{})
```

Comment:
So will we start to fully garbage collect these hashes at this point?

Reply:
`DeleteTx` is not used yet, plan is to use it in two places a) for neutrino backends where we'd rely on stored info to do naive fee bumps and b) maybe a new RPC call so the users can GC it.